Makefile

```diff
@@ -20,18 +21,21 @@ build: down
 setup:
 	docker-compose run publishing-e2e-tests bash -c 'rm -rf /app/tmp/*'
 	docker-compose up -d elasticsearch
+	docker-compose up -d mysql
```

Comment:
Did we work out if this was needed?

Reply:
We found if we didn't have it we'd get to the `db:setup` and it would say it couldn't connect. I'll give it another attempt without now and double check

Reply:
Confirmed that we have a race condition in the same way as elasticsearch, where the container is up but the service isn't ready to connect to.

``` shell
docker-compose run collections-publisher bundle exec rake db:setup                                 
Creating publishinge2etests_mysql_1              
Creating publishinge2etests_collections-publisher-worker_1                                         
I, [2017-11-09T09:58:58.629007 #1]  INFO -- sentry: ** [Raven] Raven 2.6.3 ready to catch errors   
#<Mysql2::Error: Can't connect to MySQL server on 'mysql' (111)>                                   
Couldn't create database for {"adapter"=>"mysql2", "encoding"=>"utf8", "username"=>"root", "password"=>"root", "database"=>"collections_publisher_development", "host"=>"mysql"}, {:charset=>"utf8"}   
(If you set the charset manually, make sure you have a matching collation)                         
Created database 'collections_publisher_development'                                               
#<Mysql2::Error: Can't connect to MySQL server on 'mysql' (111)>                                   
Couldn't create database for {"adapter"=>"mysql2", "encoding"=>"utf8", "username"=>"root", "password"=>"root", "database"=>"collections_publisher_test", "host"=>"mysql"}, {:charset=>"utf8"}          
(If you set the charset manually, make sure you have a matching collation)                         
Created database 'collections_publisher_test'    
rake aborted!                                    
Mysql2::Error: Can't connect to MySQL server on 'mysql' (111)                                      
/usr/local/bundle/gems/mysql2-0.4.5/lib/mysql2/client.rb:89:in `connect'                           
/usr/local/bundle/gems/mysql2-0.4.5/lib/mysql2/client.rb:89:in `initialize'                        
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/mysql2_adapter.rb:25:in `new'                                                                                          
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/mysql2_adapter.rb:25:in `mysql2_connection'                                                                            
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:729:in `new_connection'                                                                    
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:773:in `checkout_new_connection'                                                           
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:752:in `try_to_checkout_new_connection'                                                    
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:713:in `acquire_connection'                                                                
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:490:in `checkout'                                                                          
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:364:in `connection'                                                                        
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:883:in `retrieve_connection'                                                               
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_handling.rb:128:in `retrieve_connection'                                                                                        
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/connection_handling.rb:91:in `connection'                                                                                                  
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/migration.rb:1038:in `current_version' 
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/migration.rb:1255:in `last_stored_environment'                                                                                             
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/tasks/database_tasks.rb:48:in `check_protected_environments!'                                                                              
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/railties/databases.rake:11:in `block (2 levels) in <top (required)>'                                                                       
/usr/local/bundle/gems/activerecord-5.0.2/lib/active_record/railties/databases.rake:264:in `block (3 levels) in <top (required)>'                                                                      
/usr/local/bundle/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'                               
Tasks: TOP => db:schema:load => db:check_protected_environments                                    
(See full trace by running task with --trace)    
make: *** [Makefile:36: setup] Error 1
```

Reply:
argh how annoying 😞 wonder if this can be sorted by healthcheck

Reply:
That's we're hoping when we approach looking at rewriting the docker-compose to version 3

Reply:
Awesome 👍